packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -4314,3 +4354,252 @@ class InputDecorationTheme with Diagnosticable {
+      case Brightness.dark: return const Color(0x1AFFFFFF);
+      case Brightness.light:return const Color(0x0A000000) ;
+    }
+  }
+
+  @override
+  Color? get disabledFillColor {
+    // dark theme: 5% white
+    // light theme: 2% black
+    switch (Theme.of(context).brightness) {
+      case Brightness.dark:
+        return const Color(0x0DFFFFFF);
+      case Brightness.light:
+        return const Color(0x05000000) ;
+    }
+  }
+
+  @override
+  Color? get iconColor => MaterialStateColor.resolveWith((Set<MaterialState> states) {
+    if (states.contains(MaterialState.disabled) && !states.contains(MaterialState.focused)) {
+      return Theme.of(context).disabledColor;
+    }
+    if (states.contains(MaterialState.focused)) {
+      return Theme.of(context).colorScheme.primary;
+    }
+    switch (Theme.of(context).brightness) {
+      case Brightness.dark:
+        return Colors.white70;
+      case Brightness.light:
+        return Colors.black45;
+    }
+  });
+
+  @override
+  Color? get prefixIconColor => MaterialStateColor.resolveWith((Set<MaterialState> states) {
+    if (states.contains(MaterialState.disabled) && !states.contains(MaterialState.focused)) {
+      return Theme.of(context).disabledColor;
+    }
+    if (states.contains(MaterialState.focused)) {
+      return Theme.of(context).colorScheme.primary;
+    }
+    switch (Theme.of(context).brightness) {
+      case Brightness.dark:
+        return Colors.white70;
+      case Brightness.light:
+        return Colors.black45;
+    }
+  });
+
+  @override
+  Color? get suffixIconColor => MaterialStateColor.resolveWith((Set<MaterialState> states) {
+    if (states.contains(MaterialState.disabled) && !states.contains(MaterialState.focused)) {
+      return Theme.of(context).disabledColor;
+    }
+    if (states.contains(MaterialState.focused)) {
+      return Theme.of(context).colorScheme.primary;
+    }
+    switch (Theme.of(context).brightness) {
+      case Brightness.dark:
+        return Colors.white70;
+      case Brightness.light:
+        return Colors.black45;
+    }
+  });
+
+}
+
+// BEGIN GENERATED TOKEN PROPERTIES - InputDecorator
+
+// Do not edit by hand. The code between the "BEGIN GENERATED" and
+// "END GENERATED" comments are generated from data in the Material
+// Design token database by the script:
+//   dev/tools/gen_defaults/bin/gen_defaults.dart.
+
+// Token database version: v0_101
+
+// Generated version v0_101
+class _InputDecoratorDefaultsM3 extends InputDecorationTheme {
+  const _InputDecoratorDefaultsM3(this.context)
+    : super();
+
+  final BuildContext context;
+
+  @override
+  Color? get fillColor => Theme.of(context).colorScheme.surfaceVariant;
+
+  @override
+  Color? get disabledFillColor => Theme.of(context).colorScheme.onSurface.withOpacity(0.04);
+
+  @override
+  BorderSide? get activeIndicatorBorder => MaterialStateBorderSide.resolveWith((Set<MaterialState> states) {
+      if (states.contains(MaterialState.error)) {
```

Comment:
In code like this, maybe create a local variable:
```
final ColorScheme colorScheme = Theme.of(context).colorScheme;
```

Reply:
The theme code here is *generated* by gen_defaults.dart

Maybe I will update the generator script to support local variable in a separated PR.

Reply:
> This looks good and the results look good too. One bit of feedback that I didn't leave everywhere: sometimes the original defaults code stayed roughly the same, in others it applied resolveAs to a MaterialStateColor. The code might be easier to understand if we were consistent about that; it seems like in most cases we could encode the M2 defaults a MaterialStateColor, MaterialStateBorderSide, MaterialStateTextStyle.


Good point! 

So In the latest commit, I encoded most M2 default properties (colors and textStyles) as MaterialStateColor and MaterialStateTextStyle.

As for borders, _getDefaultM2BorderColor is not encoded because it is using decoration's property(decoration.hoverColor).

Reply:
Good point. Although these lookups are cheap, it might be worth updating the code generator in another PR (as you've pointed out).

Reply:
It actually already does (but perhaps not well documented 😎). Each template can define a prefix for the color scheme and text theme, which can be set to the name of a local variable defined in the body of the template:

https://github.com/flutter/flutter/blob/fa5618082db1fa0a9ac9ef987e9e210cb536fa9b/dev/tools/gen_defaults/lib/dialog_template.dart#L8-L10

and

https://github.com/flutter/flutter/blob/fa5618082db1fa0a9ac9ef987e9e210cb536fa9b/dev/tools/gen_defaults/lib/dialog_template.dart#L23-L25

Which will then do the right thing when using the `componentColor` and `textStyle` methods in the rest of the template.

Reply:
Thanks for letting me know!